### PR TITLE
feat: add service path_prefix

### PR DESCRIPTION
**Pull request by kingluo**
### Description

If `path_prefix` is specified in service, then all routes which reference this service add that prefix to uris/uri before creating the radix tree.

If `strip_path_prefix` is specified in route, then it removes the `path_prefix` before processing all phases.

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: t/node/service-path-prefix.t [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                ]]
+            ))
+            ngx.sleep(0.5)
+
+            local http = require "resty.http"
+            local httpc = http.new()
+            local uri = "http://127.0.0.1:" .. ngx.var.server_port .. "/hello"
+            local res, err = httpc:request_uri(uri)
+            ngx.status = res.status
+            ngx.log(ngx.WARN, require("inspect")(res))
```
Comment (monkeyDluffy6017) [inline t/node/service-path-prefix.t:71]: Is this for debug?

Comment (kingluo) [reply t/node/service-path-prefix.t:71]: Yes, it does not affect the test correctness.

Comment (monkeyDluffy6017) [reply t/node/service-path-prefix.t:71]: Remove it if you don't need it

File: apisix/http/route.lua [hunk 2]
```diff
@@ -31,6 +31,23 @@ local loadstring = loadstring
 local _M = {}
 
 
+function _M.prefix_uris(route, service)
+    if service and service.value.path_prefix then
+        local uri, uris
+        if route.value.uris then
+            uris = core.table.new(#route.value.uris)
+            for _, uri in ipairs(route.value.uris) do
+                core.table.insert(uris, service.value.path_prefix .. uri)
```
Comment (tokers) [inline apisix/http/route.lua:40]: A tiny detail: what if the path_prefix is ended with `/`, in such a case, we'll have continuous slashes. E.g., `/api/v1//order`. While this affects the route match?

File: apisix/init.lua [hunk 3]
```diff
@@ -640,6 +641,14 @@ function _M.http_access_phase()
             enable_websocket = service.value.enable_websocket
         end
 
+        if route.value.strip_path_prefix and service.value.path_prefix then
+            local uri = api_ctx.var.uri
+            if uri:find(service.value.path_prefix) == 1 then
```
Comment (tokers) [inline apisix/init.lua:646]: Why not using `core.string.has_prefix`?

File: docs/en/latest/admin-api.md [hunk 4]
```diff
@@ -585,6 +586,7 @@ Service resource request address: /apisix/admin/services/{id}
 | desc             | False    | Auxiliary   | Description of usage scenarios.                                                                                    | service xxxx                                     |
 | labels           | False    | Match Rules | Attributes of the Service specified as key-value pairs.                                                            | {"version":"v2","build":"16","env":"production"} |
 | enable_websocket | False    | Auxiliary   | Enables a websocket. Set to `false` by default.                                                                    |                                                  |
+| path_prefix | False    | String   | Add prefix to uri.                                                                    |                                                  |
```
Comment (monkeyDluffy6017) [inline docs/en/latest/admin-api.md:589]: Need to update the chinese doc

File: apisix/init.lua (hunk 3 above)
Comment (kingluo) [reply apisix/init.lua:646]: lua built-in string.find is more simple and high efficient.

File: apisix/http/route.lua (hunk 2 above)
Comment (kingluo) [reply apisix/http/route.lua:40]: That depends on the validation requirements.
Not only the trailing slash, the slash, and other invalid chars may exist in any position of the string. So if we need real validation, we need to define its requirement.
That is, is it enough if I just remove the trailing slash if exists?

**Comment by leslie-tsang** [comment]
Need to discuss first. :)

**Pull request closed without merging**